flink-mesos/src/main/java/org/apache/flink/mesos/entrypoint/MesosSessionClusterEntrypoint.java

```diff
@@ -113,7 +113,7 @@ protected void startClusterComponents(Configuration configuration, RpcService rp
 		RpcService rpcService,
 		HighAvailabilityServices highAvailabilityServices,
 		HeartbeatServices heartbeatServices,
-		MetricRegistry metricRegistry,
+		MetricRegistryImpl metricRegistry,
```

Comment:
This should be typed to MetricRegistry, and i suspect this also applies to many other places.

Reply:
You're right. I'll go over the different places again and try to fix it.